acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/signer/Signer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.dsl.signer;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+import static tech.pegasys.eth2signer.dsl.utils.WaitUtils.waitFor;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Signer {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final String PROCESS_NAME = "Eth2Signer";
+
+  private final Eth2SignerProcessRunner runner;
+  private final String hostname;
+  private final String urlFormatting;
+  private final Vertx vertx;
+  private HttpClient httpClient;
+
+  public Signer(final SignerConfiguration signerConfig) {
+    this.runner = new Eth2SignerProcessRunner(signerConfig);
+    this.hostname = signerConfig.hostname();
+    urlFormatting = "http://%s:%s";
+    vertx = Vertx.vertx();
```

Comment:
does it make sense to inject vertx instance via constructor? ... I recall some discussion around using single instance so it might come handy later on.

Reply:
given how our AT work - there is only ever 1 Signer (i.e. they don't interact with each other) - thus there will only ever be one of these in a given test.

Reply:
👍